internal/models/source_model.go

```diff
@@ -0,0 +1,23 @@
+package models
+
+type Source struct {
+	// Required auto-generated PK.
+	ID uint64 `db:"id" json:"id"`
+
+	// Source ID in sources. Unique.
+	SourceID uint64 `db:"source_id" json:"source_id"`
+
+	// User-facing name. Required.
+	Name string `db:"name" json:"name"`
+
+	// Tenant ID. Required.
+	TenantID uint64 `db:"tenant_id" json:"tenant_id"`
+
+	// Associated Account model. Required.
+	AccountId uint64 `db:"account_id" json:"account_id"`
+
+	// Authentication ID in sources lined to a spesific source in order to fetch the ARN
+	AuthID string `db:"auth_id" json:"auth_id"`
```

Comment:
And this is their text-id if I am not mistaken: https://github.com/RedHatInsights/sources-api-go/blob/main/model/authentication.go#L17

I am fine with storing this as text if that's what they recommend.

Reply:
it is stored here as a text

Reply:
Actually from the code it looks like authid is actually UUID. I will ask on the channel for confirmation, but if that's the case we could save ton of data by using the native UUID postgres type. Unfortunately the standard driver does only support storing UUID as text or 16 bytes, however, native pgx driver would allow us to use the native type.

I am making a note for my "Switch to Scanny/PGX driver" proposal about this, that is significant saving for data and indices. Let's keep it as string for now.